components/admin_console/custom_terms_of_service_settings/custom_terms_of_service_settings.tsx

```diff
@@ -15,27 +16,40 @@ import LoadingScreen from 'components/loading_screen';
 
 import {Constants} from 'utils/constants';
 
-export default class CustomTermsOfServiceSettings extends AdminSettings {
-    static propTypes = {
-        actions: PropTypes.shape({
-            getTermsOfService: PropTypes.func.isRequired,
-            createTermsOfService: PropTypes.func.isRequired,
-        }).isRequired,
-        config: PropTypes.object,
-        license: PropTypes.object,
-        setNavigationBlocked: PropTypes.func,
-
-        /*
-        * Action to save config file
-        */
-        updateConfig: PropTypes.func,
+type Props = BaseProps & {
+    actions: {
+        getTermsOfService: () => Promise<{data: TermsOfService}>;
+        createTermsOfService: (text: string) => Promise<{data: TermsOfService, error?: Error}>;
     };
+    config: DeepPartial<AdminConfig>;
```

Comment:
Why partial and not directly AdminConfig?

Reply:
This used partial initially to keep the test simple, lets see how it looks with the test adjusted instead.